### PR TITLE
[fix] exclude emitted declarations on packaging

### LINKED ISSUE #1923: emit_dts in svelte-kit package should respect the file excludes config
### Describe the problem

It's fantastic that automatic type generation is included with the `package` command, however it seems `svelte2tsx` [doesn't accept any other configuration besides the lib root directory](https://github.com/sveltejs/language-tools/blob/master/packages/svelte2tsx/src/emitDts.ts#L8) so when the [types are generated](https://github.com/sveltejs/kit/blob/1f8abd9aa3f244c1011f2b65a5d20d489989c319/packages/kit/src/core/make_package/index.js#L17) with `svelte-kit package`, all files get generated types.

This is not desired when `.stories.svelte`, `.test.ts` or `.test.svelte` files (files that [help test slots as a workaround](https://svelte-recipes.netlify.app/testing/#testing-slots)) are next to the `.svelte` files in the `lib` directory.

### Describe the proposed solution

It would be great to have the file excludes respected during type generation. I understand that it might be `svelte2tsx` that needs to be altered to make this work.

### Alternatives considered

For now I think I'll include a `post-package` script that cleans up files that match the glob in `excludes`. It's a hacky workaround, why generate files that need to be cleaned up, right?

### Importance

would make my life easier

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ignatiusmb**
This PR contains fixes for a couple of things, including but not limited to

- properly excluding files as well as its generated declarations
- improvements for the globbing pattern users can provide
- nested files not included/excluded on windows machines

Changing our globbing package allows end-users to provide more complex but easier (more familiar, bash-like) to write patterns. It's not as tiny in size as the previous one, but we're only using it for the packaging cli, and the patterns it allows our users to write also relieves a lot of normalizing and handling on our side as well.

Fixes #1923, waiting #2236 for this to be reviewed

### CODE REVIEW

File: packages/kit/src/packaging/index.js [hunk 1]
```diff
@@ -21,8 +21,8 @@ export async function make_package(config, cwd = process.cwd()) {
 
 	const files_filter = create_filter(config.kit.package.files);
 	const exports_filter = create_filter({
-		...config.kit.package.exports,
-		exclude: [...config.kit.package.exports.exclude, '*.d.ts']
+		include: config.kit.package.exports.include,
```
Comment (dummdidumm) [inline packages/kit/src/packaging/index.js:24]: Why does this need to be set explicitely? Isn't it part of the spread already?

**Review by dummdidumm — APPROVED** [review]

File: packages/kit/src/packaging/index.js (hunk 1 above)
Comment (ignatiusmb) [reply packages/kit/src/packaging/index.js:24]: Always found myself taking a while to read and understand when I get to this part. Explicitly setting the include option makes it clearer as it now has the same indentation, and avoids spreading and immediately overriding the exclude values.

Comment (dummdidumm) [reply packages/kit/src/packaging/index.js:24]: Makes sense 👍

**Comment by ignatiusmb** [comment]
This may also remove handwritten `.d.ts` that are not part of the exclude pattern, but then again, it doesn't really make sense to publish the typings without the logic. Except for a different case when it only contains `export type`, in which case users should make sure it is isolated properly and won't be caught by their glob pattern, or rely on `.npmignore`.

I wanted to add some disclaimer in the docs that users are fully responsible for their glob patterns when they're using the include/exclude options. But, those who are using these options should have known better themselves. I guess we'll have to see if there's any complaint later on.

**Pull request merged**